mecanic/client.lua

```diff
@@ -0,0 +1,110 @@
+local Dialog = ImportPackage("dialogui")
+local _ = function(k,...) return ImportPackage("i18n").t(GetPackageName(),k,...) end
+
+local mecanicNPC
+local mecanicNpcMenu
+local mecanicMenu
+local colorMenu
+
+AddRemoteEvent("Setupmecanic", function(mecanicnpc) 
+    mecanicNPC = mecanicnpc
+end)
+
+AddEvent("OnTranslationReady", function()
+    mecanicNpcMenu = Dialog.create(_("mecanic_menu"), nil, _("start_stop_mecanic") ,_("cancel"))
+    mecanicMenu = Dialog.create(_("mecanic_menu"), nil, _("RepaireMecanic"), _("ColorMecanic"), _("RefuelMecanic"), _("cancel"))
+    colorMenu = Dialog.create(_("color_menu"), nil, _("black"), _("white"), _("blue"), _("green"), _("red"), _("yellow"), _("purple"), _("cyan"), _("cancel"))
+end)
+
+AddEvent("OnKeyPress", function( key )
+    if key == "E" and not onSpawn and not onCharacterCreation then
+        local Nearestmecanic = GetNearestmecanic()
+        if Nearestmecanic ~= 0 then
+            local ScreenX, ScreenY = GetScreenSize()
+            SetMouseLocation(math.floor(ScreenX / 2), math.floor(ScreenY / 2))
+            Dialog.show(mecanicNpcMenu)
+		end
+    end
+    if key == "F3" and not onSpawn and not onCharacterCreation then
+        local ScreenX, ScreenY = GetScreenSize()
+        SetMouseLocation(math.floor(ScreenX / 2), math.floor(ScreenY / 2))
+        CallRemoteEvent("OpenmecanicMenu")
+    end
+end)
+
+
+AddEvent("OnDialogSubmit", function(dialog, button, ...)
+	if dialog == mecanicNpcMenu then
+		if button == 1 then
+            CallRemoteEvent("StartStopmecanic")
+        end
+    end
+    if dialog == mecanicMenu then
+        if button == 1 then
+            CallRemoteEvent("RepaireMecanic")
+        end
+        if button == 2 then
+            CallRemoteEvent("ColorMecanic")
+        end
+        if button == 3 then
+            CallRemoteEvent("RefuelMecanic")
+        end
+    end
+    if dialog == colorMenu then
+        if button == 1 then
+            CallRemoteEvent("black")
```

Comment:
Mayne rename the event to `changeVehicleColor` and send the color as an argument to this function?

Reply:
Comment ça ? @rdlh